### ISSUE TITLE
Having Blue Shift maps in the maps directory causes the engine to think other maps are Blue Shift maps

### ISSUE THREAD

**Issue by kargaroc** [FWGS/xash3d-fwgs#665]
The code that loads Blue Shift maps does the following:
```
in:
engine/common/mod_bmodel.c -> Mod_LoadBmodelLumps
engine/common/mod_bmodel.c -> Mod_TestBmodelLumps

if( header->version == HLBSP_VERSION && header->lumps[LUMP_ENTITIES].fileofs <= 1024 && (header->lumps[LUMP_ENTITIES].filelen % sizeof( dplane_t )) == 0 ) {
	// blue-shift swapped lumps
	srclumps[0].lumpnumber = LUMP_PLANES;
	srclumps[1].lumpnumber = LUMP_ENTITIES;
}
```

Basically, if the map is a Half life map, and some stuff in the header is weird, it assumes its a Blue Shift map and makes the necessary changes.
But, the data in `srclumps` looks like its global, so its used for all maps, not just the map we're checking.
As a result, it believes that *all* maps encountered afterwards are Blue Shift maps, regardless of their format.
This breaks things.
When playing Blue Shift this is never encountered, since all the maps in the Blue Shift maps directory are like this.

The proper thing to do here is probably to refactor things such that `srclumps` is local to the map in question.
But, this code always runs when loading a map, so it can be written as the following:
```
if( header->version == HLBSP_VERSION ) {
	// only relevant for half-life maps
	if( header->lumps[LUMP_ENTITIES].fileofs <= 1024 && (header->lumps[LUMP_ENTITIES].filelen % sizeof( dplane_t )) == 0 ) {
		// blue-shift swapped lumps
		srclumps[0].lumpnumber = LUMP_PLANES;
		srclumps[1].lumpnumber = LUMP_ENTITIES;
	} else {
		// everything else
		srclumps[0].lumpnumber = LUMP_ENTITIES;
		srclumps[1].lumpnumber = LUMP_PLANES;
	}
}
```
This just fixes the settings if its not a Blue Shift map, which makes other maps work.
Probably still bugs but I've not encountered any with this fix.

**Comment by a1batross** [comment]
Thanks for the report!

Indeed, looks very suspicious. To me, your fix looks correct. Could you send a PR?

**Comment by a1batross** [comment]
Except I would restore srclumps in else branch of BSP version check.

**Comment by a1batross** [comment]
Fixed in 24edcf7
